VocaDbModel/Helpers/MailAddressNormalizer.cs

```diff
@@ -0,0 +1,136 @@
+			// Microsoft
+			{ "outlook.com", '+' },
+			{ "hotmail.com", '+' },
+			{ "live.com", '+' },
+			// Fastmail - https://www.fastmail.com/help/receive/addressing.html TODO: whatever@username.fastmail.com -> username@fastmail.com
+			{ "fastmail.com", '+' },
+			{ "fastmail.fm", '+' },
+			// Yahoo Mail Plus accounts, per https://en.wikipedia.org/wiki/Yahoo!_Mail#Email_domains, use hyphens - http://www.cnet.com/forums/discussions/did-yahoo-break-disposable-email-addresses-mail-plus-395088/
+			{ "yahoo.com.ar", '-' },
+			{ "yahoo.com.au", '-' },
+			{ "yahoo.at", '-' },
+			{ "yahoo.be/fr", '-' },
+			{ "yahoo.be/nl", '-' },
+			{ "yahoo.com.br", '-' },
+			{ "ca.yahoo.com", '-' },
+			{ "qc.yahoo.com", '-' },
+			{ "yahoo.com.co", '-' },
+			{ "yahoo.com.hr", '-' },
+			{ "yahoo.cz", '-' },
+			{ "yahoo.dk", '-' },
+			{ "yahoo.fi", '-' },
+			{ "yahoo.fr", '-' },
+			{ "yahoo.de", '-' },
+			{ "yahoo.gr", '-' },
+			{ "yahoo.com.hk", '-' },
+			{ "yahoo.hu", '-' },
+			{ "yahoo.co.in/yahoo.in", '-' },
+			{ "yahoo.co.id", '-' },
+			{ "yahoo.ie", '-' },
+			{ "yahoo.co.il", '-' },
+			{ "yahoo.it", '-' },
+			{ "yahoo.co.jp", '-' },
+			{ "yahoo.com.my", '-' },
+			{ "yahoo.com.mx", '-' },
+			{ "yahoo.ae", '-' },
+			{ "yahoo.nl", '-' },
+			{ "yahoo.co.nz", '-' },
+			{ "yahoo.no", '-' },
+			{ "yahoo.com.ph", '-' },
+			{ "yahoo.pl", '-' },
+			{ "yahoo.pt", '-' },
+			{ "yahoo.ro", '-' },
+			{ "yahoo.ru", '-' },
+			{ "yahoo.com.sg", '-' },
+			{ "yahoo.co.za", '-' },
+			{ "yahoo.es", '-' },
+			{ "yahoo.se", '-' },
+			{ "yahoo.ch/fr", '-' },
+			{ "yahoo.ch/de", '-' },
+			{ "yahoo.com.tw", '-' },
+			{ "yahoo.co.th", '-' },
+			{ "yahoo.com.tr", '-' },
+			{ "yahoo.co.uk", '-' },
+			{ "yahoo.com", '-' },
+			{ "yahoo.com.vn", '-' }
+		};
+
+		public static string Normalize(MailAddress address, MailAddressNormalizerOptions options = MailAddressNormalizerOptions.None) => NormalizeAsync(address, options).Result;
+
+		public static string Normalize(string address, MailAddressNormalizerOptions options = MailAddressNormalizerOptions.None) => NormalizeAsync(address, options).Result;
+
+		public static async Task<string> NormalizeAsync(MailAddress address, MailAddressNormalizerOptions options = MailAddressNormalizerOptions.None) {
+			var user = address.User;
+			var host = address.Host.ToLower();
+
+			if (options.HasFlag(MailAddressNormalizerOptions.ForceRemoveTags))
+				user = Regex.Replace(user, @"[-+=].*", "");
+			else {
+				if (hostsWithTags.TryGetValue(host, out var separator))
+					user = user.Split(separator)[0];
+			}
+
+			if (options.HasFlag(MailAddressNormalizerOptions.ForceRemoveDots) || Regex.IsMatch(host, @"^(gmail|googlemail|google)\.com$"))
+				user = Regex.Replace(user, @"\.", "");
+
+			if (host == "googlemail.com")
+				host = "gmail.com";
+
+			if (options.HasFlag(MailAddressNormalizerOptions.DetectProvider)) {

```

Comment:
I think we don't actually need this, for now at least. It's unlikely the spammers would go through the trouble of custom domain, at least so far they haven't. It's fine to keep it simple for now. In my opinion, it might be better remove the code if it doesn't even work properly.